File: 01-BACKEND/spring-boot-library/src/main/java/com/luv2read/springbootlibrary/requestModels/AddBookRequest.java
package com.luv2read.springbootlibrary.requestModels;

import lombok.Data;

@Data
public class AddBookRequest {

    // Title of the book
    private String title;

    // Author of the book
    private String author;

    // Description of the book
    private String description;

    // Category of the book
    private String category;

    // URL or path to the book's image
    private String img;

    // Total number of copies for the book
    private int copies;

    // Parameterized constructor for creating AddBookRequest objects with book details
    public AddBookRequest(String title, String author, String description, String category, String img, int copies) {
        this.title = title;
        this.author = author;
        this.description = description;
        this.category = category;
        this.img = img;
        this.copies = copies;
    }
}

File: 02-FRONTEND/react-library/src/models/AddBookRequest.ts
// Class representing the request model for adding a book
class AddBookRequest {
    // Properties representing book details
    title: string;
    author: string;
    description: string;
    copies: number;
    category: string;
    img?: string; // Optional property for book image URL

    // Constructor to initialize the book details
    constructor(
        title: string,
        author: string,
        description: string,
        copies: number,
        category: string,
    ) {
        this.title = title;
        this.author = author;
        this.description = description;
        this.copies = copies;
        this.category = category;
    }
}

// Exporting the AddBookRequest class
export default AddBookRequest;
